### PR TITLE
TCM-104 | Build dummy update screen 

### PR DESCRIPTION
**Pull request by halilyesilyurt**
### Summary

- [x] Completed dummy update screen.
- [x] Added update screen navigation to test utility.

### Issue

[TCM-104](https://hyfnla.atlassian.net/browse/TCM-104)

### Video/Screenshot

| Create| Edit|
|--------|--------|
|Create|Edit|

### CODE REVIEW

File: app/src/main/res/layout/fragment_update.xml [hunk 1]
```diff
@@ -0,0 +1,200 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_update.xml:1]: Some extra spaces in this file

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -24,18 +24,18 @@
     <string name="home_on_demand_title_label">ON-DEMAND HOT &amp; NEW</string>
     <string name="home_upcoming_button_label">VIEW ALL →</string>
 
-    <string name="inputview_placeholder_birthday">Birthday</string>
+    <string name="inputview_placeholder_birthday">BIRTHDAY</string>
```
Comment (VivekV95) [inline app/src/main/res/values/strings.xml:27]: Let's keep the strings consistent with [iOS](https://github.com/hyfn/trx-consumer-ios/blob/develop/TRX-Consumer/Screens/Input/InputViewState.swift) for now

File: app/src/main/res/layout/fragment_update.xml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            android:scaleType="centerInside"
+            android:src="@drawable/ic_btn_back"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+    </com.trx.consumer.common.CommonView>
+
+    <ScrollView
+        android:layout_width="0dp"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_update.xml:33]: Can just make width `match_parent` and remove the horizontal constraints

File: app/src/main/res/layout/fragment_update.xml (hunk 1 above)
Comment (VivekV95) [reply app/src/main/res/layout/fragment_update.xml:1]: Let's also code format this file

File: app/src/main/res/layout/fragment_update.xml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+<com.trx.consumer.common.CommonView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    app:backgroundColor="@color/white">
+
+    <com.trx.consumer.common.CommonView
+        android:id="@+id/viewNavBar"
+        android:layout_width="0dp"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_update.xml:11]: Same here, can make width `match_parent` and remove horizontal constraints

File: app/src/main/res/layout/fragment_update.xml [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+                android:id="@+id/ivFirstName"
+                android:layout_width="0dp"
+                android:layout_height="60dp"
+                android:layout_marginTop="30dp"
+                app:inputViewState="first"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/ivProfile" />
+
+            <View
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_update.xml:64]: Let's make all the dividers `greyLight`

File: app/src/main/res/layout/fragment_update.xml [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/viewNavBar">
+
+
+        <com.trx.consumer.common.CommonView
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content">
+
+            <com.trx.consumer.common.CommonImageView
+                android:id="@+id/ivProfile"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_update.xml:46]: Convention is `imgProfile`

File: app/src/main/res/layout/fragment_update.xml [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+
+    <com.trx.consumer.common.CommonButton
+        android:id="@+id/btnSave"
+        android:layout_width="0dp"
+        android:layout_height="50dp"
+        android:layout_marginStart="16dp"
+        android:layout_marginEnd="16dp"
+        android:layout_marginBottom="32dp"
+        android:fontFamily="@font/atcarquette_regular"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_update.xml:178]: Specs have this as regular but let's make this bold like iOS is doing, we'll probably eventually create a style for this

File: app/src/main/res/layout/fragment_update.xml [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        tools:text="@string/update_save_label" />
+
+
+    <com.trx.consumer.common.CommonHudView
+        android:id="@+id/hudView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:visibility="gone"
+        app:layout_constraintBottom_toBottomOf="parent"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_update.xml:195]: Can use `style="@style/MatchParentEdgesViewStyle"` for this

File: app/src/main/res/layout/fragment_update.xml [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+
+        </com.trx.consumer.common.CommonView>
+
+
+    </ScrollView>
+
+
+    <com.trx.consumer.common.CommonButton
+        android:id="@+id/btnSave"
+        android:layout_width="0dp"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_update.xml:173]: Can `match_parent` here and remove horizontal constraints

File: app/src/main/java/com/trx/consumer/views/input/InputViewState.kt [hunk 10]
```diff
@@ -127,7 +127,11 @@ enum class InputViewState(val placeholder: Int, vararg val type: Int) {
         textView.text?.let { userInput ->
             when (userInput.length) {
                 1 -> {
-                    if (userInput.first().toString().toInt() !in intArrayOf(0, 1) && before < count) {
+                    if (userInput.first().toString().toInt() !in intArrayOf(
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/views/input/InputViewState.kt:130]: Let's revert this, auto-format probably picked it up but we'll dedicate tickets to fixing core files

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Looking great, left a few comments

File: app/src/main/res/layout/fragment_update.xml (hunk 1 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_update.xml:1]: Completed.

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (halilyesilyurt) [reply app/src/main/res/values/strings.xml]: Completed.

File: app/src/main/res/layout/fragment_update.xml (hunk 3 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_update.xml]: Completed.

File: app/src/main/res/layout/fragment_update.xml (hunk 4 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_update.xml]: Completed.

File: app/src/main/res/layout/fragment_update.xml (hunk 5 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_update.xml:59]: Completed.

File: app/src/main/res/layout/fragment_update.xml (hunk 6 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_update.xml]: Completed.

File: app/src/main/res/layout/fragment_update.xml (hunk 7 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_update.xml]: Completed.

File: app/src/main/res/layout/fragment_update.xml (hunk 8 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_update.xml]: Completed.

File: app/src/main/res/layout/fragment_update.xml (hunk 9 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_update.xml]: Completed.

File: app/src/main/java/com/trx/consumer/views/input/InputViewState.kt (hunk 10 above)
Comment (halilyesilyurt) [reply app/src/main/java/com/trx/consumer/views/input/InputViewState.kt]: Completed.

**Review by VivekV95 — APPROVED** [review]

**Comment by onodude** [comment]
we wont have the profile image and some paddings/colors seem off lets check iOS for parity and also resolve the conflicts

**Comment by halilyesilyurt** [comment]
> we wont have the profile image and some paddings/colors seem off lets check iOS for parity and also resolve the conflicts

removed profile image and tried to make similar to ios side.

**Comment by halilyesilyurt** [comment]
pr images updated.

**Review by VivekV95 — APPROVED** [review]

**Review by onodude — APPROVED** [review]

**Pull request merged**